stripe/src/main/java/com/stripe/android/paymentsheet/BaseAddCardFragment.kt

```diff
@@ -181,19 +182,24 @@ internal abstract class BaseAddCardFragment(
     }
 
     private fun updateSelection() {
-        sheetViewModel.updateSelection(
-            if (addCardViewModel.isCardValid) {
-                paymentMethodParams?.let { params ->
-                    PaymentSelection.New.Card(
-                        params,
-                        cardMultilineWidget.brand,
-                        shouldSavePaymentMethod = shouldSaveCard()
-                    )
-                }
-            } else {
-                null
+        val validCard = if (addCardViewModel.isCardValid) {
+            paymentMethodParams?.let { params ->
+                PaymentSelection.New.Card(
+                    params,
+                    cardMultilineWidget.brand,
+                    shouldSavePaymentMethod = shouldSaveCard()
+                )
             }
-        )
+        } else {
+            null
+        }
+
+        // If you open a new unsaved card, edit it, go to the list view and come back the edited
+        // card should be shown, this means that the new card must be updated
+        validCard?.let {
```

Comment:
This handles the following scenario(s):
1. In payment custom (options) or payment complete with a customer
2. Add a new card (save or don't save), and make sure it is valid (don't press add button)
3. Press back to see the list again
4. Press + to add a new card, and you should see the fields.
5. Edit the card and make sure it is valid (don't press add button)
6. Press back to see the list again.
7. Press + to add a new card, and you should see all the fields populated again with your latest edits.